### DISCUSSION TITLE
Using this library with Terramaster NAS

### DISCUSSION THREAD

**Discussion by jparchib** [dgreif/ring#845] [Q&A] [answered]
Hi, i'm pretty new to this and although I have a background in programming, it's from a very long time ago.  Do you think it's possible to have these libraries to work on a terramaster NAS?  And if so, is there some simple script I can look for to get me started?  What I want is to be able to record the stream when the ring doorbell detech motion and have it saved locally on the NAS as a video file.

Is it possible?

Thanks!

**Comment by dgreif** [comment, marked as answer]
That should definitely be possible.  You _could_ use ffmpeg to stream directly to a file on your NAS using the `recordToFile` method on the camera.  See https://github.com/dgreif/ring/blob/master/examples/record-example.ts for an example.  If you have a Ring subscription, you could also let them record the clips on motion, and then download them later on.  Either approach would take some code to get set up and watch for events (Something like https://github.com/dgreif/ring/blob/master/examples/example.ts#L71-L85)
